aten/src/ATen/native/Constraints.cpp

```diff
@@ -20,16 +21,41 @@
 namespace at {
 namespace native {
 
-void sym_constrain_range_cpu(
+void sym_constrain_range(
     const Scalar& size,
     c10::optional<int64_t> min,
-    c10::optional<int64_t> max) {}
+    c10::optional<int64_t> max) {
 
-Tensor _functional_sym_constrain_range_cpu(
+    int64_t min_val = min.has_value() ? min.value() : 0;
```

Comment:
This is flat out wrong. If min is not specified, the minimum is unbounded below (could be negative). You need to simply skip the le check if it is none. Ditto below (where it is correct, but it will be more symmetric to skip the upper bound test when it's none.)